src/server/slowlog.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <boost/circular_buffer.hpp>
+#include <string>
+#include <vector>
+
+#include "base/integral_types.h"
+#include "server/common.h"
+namespace dfly {
+
+class SlowlogEntry {
+ public:
+  SlowlogEntry(uint64_t command_id, uint64_t unix_timestamp, uint64_t execution_time_micro,
+               std::vector<std::string> cmd_args, std::string client_ip, std::string client_name);
+
+  const std::vector<std::string>& cmdArgs() const {
+    return cmd_args_;
+  }
+  int64 commandId() const {
+    return command_id_;
+  }
+  int64 unixTimestamp() const {
+    return unix_timestamp_;
+  }
+  int64 executionTimeMicro() const {
+    return execution_time_micro_;
+  }
+  const std::string& clientIp() const {
+    return client_ip_;
+  }
+  const std::string& clientName() const {
+    return client_name_;
+  }
+
+ private:
+  std::vector<std::string> cmd_args_;
+  uint64_t command_id_;
+  uint64_t unix_timestamp_;
+  uint64_t execution_time_micro_;
+  std::string client_ip_;
+  std::string client_name_;
+};
+
+class SlowlogData {
+ public:
+  SlowlogData() = default;
+  SlowlogData(int64 slowlogEntryId, int totalThreads)
```

Comment:
you do not really need deducing a global id inside `SlowlogData`. btw, maybe `SlowLogShard` is a better name. I do not think you need `slowlogEntryId` and  `totalThreads` and compute non colliding ids at each shard level. Instead you can maintain a single counter per each shard and do the translation inside the SlowLog command itself when you fetch and fill all the entries. 
There - (id 1, shard 0) will be translated to 8 while (id 1, shard 1) will be translated to 9 in 8-shard system similarly to how you do it today. This will decrease complexity and will make this change less intrusive, imho.

Reply:
Yep, your suggested approach is much more elegant.

> btw, maybe SlowLogShard is a better name.

Changed this.